tutorials/ner_pytorch_medical/scripts/azure/azure_ner_pipe.py

```diff
@@ -0,0 +1,136 @@
+    ):
+        """Initialize AzureEntityRecognizer
+        nlp (Language): spaCy Language Object
+        client (TextAnalyticsClient): Text Analytics client
+        extension_attr (str): Attribute name for the extension property to set the ents to
+        use_extension_attr (bool): Use extension attr or set to doc.ents directly if False
+        name (str): The component instance name.
+        """
+        self.name = name
+        self.client = client
+        self.use_extension_attr = use_extension_attr
+        if self.use_extension_attr:
+            self.extension_attr = extension_attr
+            Doc.set_extension(self.extension_attr, default=[])
+
+    def __call__(self, doc: Doc) -> Doc:
+        """Extract entities from a single document using Azure Text Analytics API
+        doc (Doc): Input doc
+        RETURNS (Doc): Doc with extracted entities set
+        """
+        preds = self.predict([doc])
+        self.set_annotations([doc], preds)
+        return doc
+
+    def pipe(self, stream: Iterable[Doc], *, batch_size: int = 128) -> Iterator[Doc]:
+        """Extract entities from a stream of documents using Azure Text Analytics API.
+
+        stream (Iterable[Doc]): A stream of documents.
+        batch_size (int): The number of documents to buffer.
+        YIELDS (Doc): Processed documents in order.
+        """
+        for docs in util.minibatch(stream, size=batch_size):
+            preds = self.predict(docs)
+            self.set_annotations(docs, preds)
+            yield from docs
+
+    def predict(self, docs: Iterable[Doc]) -> Iterable[ResponseDocument]:
+        """Extract entities from a batch of documents using Azure Text Analytics API.
+
+        docs (Iterable[Doc]): The documents to predict.
+        RETURNS: Raw predictions from Azure Text Analytics API
+        """
+        res = self.client.predict([d.text for d in docs])
+        return res.documents
+
+    def set_annotations(self, docs: Iterable[Doc], preds: Iterable[ResponseDocument]):
+        """Modify a batch of documents with predictions from the Azure Text Analytics API
+
+        docs (Iterable[Doc]): The documents to modify.
+        preds (Iterable[ResponseDocument]): Predictions from Azure
+        """
+        for doc, pred in zip(docs, preds):
+            spacy_ents = []
+            for entity in pred.entities:
+                start = entity.offset
+                end = entity.offset + entity.length
+                label = entity.category
+                spacy_ents.append(doc.char_span(start, end, label=label))
+
+            if self.use_extension_attr:
+                setattr(doc._, self.extension_attr, spacy_ents)
+            else:
+                # Note this does not currently preserve entities previously set in the pipeline.
+                # So you would want to run this before an existing ner pipeline component.
+                # e.g. nlp.add_pipe("azure_ner", before="ner")
+                doc.ents = spacy_ents
```

Comment:
It might make sense to raise a warning here if there are already some entities defined on the doc, so the user gets alerted to the problem & possible fix (reordering)

Reply:
Sure thing, I'll add this.